SonarQube.MSBuild.Tasks/Targets/SonarQube.Integration.targets

```diff
@@ -188,8 +193,10 @@
        otherwise those results will not appear in the project info file.
   -->
   <!-- **************************************************************************** -->
-  <Target Name="WriteSonarQubeProjectData" AfterTargets="Build"
-        Condition=" $(SonarQubeTempPath) != '' ">
+  <Target Name="WriteSonarQubeProjectData"
+          DependsOnTargets="SonarQubeCategoriseProject;CalculateSonarQubeFilesToAnalyze"
+          AfterTargets="Build"
+          Condition=" $(SonarQubeTempPath) != '' ">
```

Comment:
The "write project" target uses properties set in the helper targets so it lists them as dependencies.
We are using "AfterTargets" to specify when this target should be invoked in the overall process.